src/jit/codegencommon.cpp

```diff
@@ -5653,6 +5654,15 @@ void          CodeGen::genPushCalleeSavedRegisters()
     //
     rsPushRegs |= RBM_LR; // We must save the return address (in the LR register)
 
+#if defined(_TARGET_ARM_)
+    if (compiler->compLocallocUsed)
+    {
+        // Don't forget store REG_SAVED_LOCALLOC_SP previous value
+        // since it will be trashed with SP
+        rsPushRegs |= RBM_SAVED_LOCALLOC_SP;
+    }
+#endif // _TARGET_ARM_
```

Comment:
This is the wrong place for this. Instead, add the following to CodeGen::genFinalizeFrame() 

```
#if defined(_TARGET_ARM_)
        // If there are any reserved registers, add them to the 
        if (regSet.rsMaskResvd != RBM_NONE)
        {
            regSet.rsSetRegsModified(regSet.rsMaskResvd);
        }
#endif // _TARGET_ARM_
```
(just before:
```
#ifdef DEBUG
    if (verbose)
    {
        printf("Modified regs: ");
```

The idea is that REG_SAVED_LOCALLOC_SP is already correctly in the rsMaskResvd mask.

The legacy JIT does this during code generation, but RyuJIT needs to do this here instead. (It appears to be ok, and cause zero diffs, to do this for legacy JIT as well.)